### PR TITLE
[AccountManager] Improving resource handling - Issues 67,68 & 69

### LINKED ISSUE #67: [AccountManger] Wrong Implement of Dispose pattern
https://github.com/Samsung/TizenFX/blob/3566375f1189cdcd03f869fae78eabd77bbe3a3b/src/Tizen.Account.AccountManager/Tizen.Account.AccountManager/Account.cs#L624-L630

When disposing was true, 
 - Unmanaged and managed resource should be released. 

When disposing was false, 
 - Only Unmanaged resource should be released.

And SafeHandle is managed resource

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/dispose-pattern

### LINKED ISSUE #68: [AccountManager] Wrong implement of SafeHandle.ReleaseHandle
https://github.com/Samsung/TizenFX/blob/3566375f1189cdcd03f869fae78eabd77bbe3a3b/src/Tizen.Account.AccountManager/Tizen.Account.AccountManager/SafeAccountHandle.cs#L61-L65

You need call native handle release function

https://msdn.microsoft.com/ko-kr/library/system.runtime.interopservices.safehandle(v=vs.110).aspx

Nobody call native release function
https://github.com/Samsung/TizenFX/blob/3566375f1189cdcd03f869fae78eabd77bbe3a3b/src/Tizen.Account.AccountManager/Interop/Interop.Account.cs#L36-L37

### LINKED ISSUE #69: [AccountManager] Wrong ownership handing
https://github.com/Samsung/TizenFX/blob/3566375f1189cdcd03f869fae78eabd77bbe3a3b/src/Tizen.Account.AccountManager/Tizen.Account.AccountManager/AccountService.cs#L107-L113

A handle in foreach callback of native, will released when callback was completed, 
So, You didn't got a handle ownership. It will be released by native.
If you release the handle, it is a double free

### PR DESCRIPTION
**Pull request by abhishekvijay**
Signed-off-by: Abhishek Vijay <abhishek.v@samsung.com>

### Description of Change ###
Resource handling issues fixed for AccountManager. 

### Issues Resolved ###
These fixes are with respect to following issues:
- fixes #67 
- fixes #68
- fixes #69

### CODE REVIEW

File: src/Tizen.Account.AccountManager/Tizen.Account.AccountManager/Account.cs [hunk 1]
```diff
@@ -623,10 +623,14 @@ public void Dispose()
...
         {
-            if (!disposing)
-            {
-                _handle.Dispose();
-            }
+		if (disposing)
+		{
+			if (_handle != IntPtr.Zero)
+			{
+				Interop.Account.Destroy(_handle);
```
Comment (myroot) [inline src/Tizen.Account.AccountManager/Tizen.Account.AccountManager/Account.cs]: _handle is managed object (safehandle)
So, you better call _hanlde.Dispose()

File: src/Tizen.Account.AccountManager/Tizen.Account.AccountManager/AccountService.cs [hunk 2]
```diff
@@ -108,7 +108,6 @@ public static IEnumerable<Account> GetAccountsAsync()
             {
                 Account account = new Account(new SafeAccountHandle(data, true));
                 values.Add(account.AccountId);
-                account.Dispose();
```
Comment (myroot) [inline src/Tizen.Account.AccountManager/Tizen.Account.AccountManager/AccountService.cs]: Why do you remove Dispose call?

**Comment by myroot** [comment]
https://github.com/Samsung/TizenFX/issues/68
is not fixed by this pr

File: src/Tizen.Account.AccountManager/Tizen.Account.AccountManager/AccountService.cs (hunk 2 above)
Comment (myroot) [reply src/Tizen.Account.AccountManager/Tizen.Account.AccountManager/AccountService.cs]: Actually, SafeAccountHandle has no ownership
(data is not vaild outside of callback)

so it should be like below
``` c#
new SafeAccountHandle(data, false);
```

File: src/Tizen.Account.AccountManager/Tizen.Account.AccountManager/Account.cs [hunk 3]
```diff
@@ -623,10 +623,10 @@ public void Dispose()
         /// <param name="disposing">The boolean value for destoying account handle.</param>
         protected virtual void Dispose(bool disposing)
         {
-            if (!disposing)
-            {
-                _handle.Dispose();
-            }
+		if (disposing)
```
[the remark this answers is not in the document]
Comment (abhishekvijay) [reply src/Tizen.Account.AccountManager/Tizen.Account.AccountManager/Account.cs]: All tabs are removed from the files.

**Pull request merged**